src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -187,87 +182,81 @@ public void Dispose()
         }
 
         /// <summary>
-        /// This function will evaluate given string and it will try to convert
-        /// it to a Uri object. The string could contain URI fragment, IP address and  port
-        /// tuple or just IP address or name. It will return null if parsing fails.
+        /// This function is used to parse WinINet Proxy strings. The strings are a semicolon
+        /// or whitespace separated list, with each entry in the following format:
+        /// ([&lt;scheme&gt;=][&lt;scheme&gt;"://"]&lt;server&gt;[":"&lt;port&gt;])
         /// </summary>
-        private static Uri GetUriFromString(string value)
+        private static void ParseProxyConfig(string value, out Uri insecureProxy, out Uri secureProxy)
         {
+            insecureProxy = null;
+            secureProxy = null;
             if (string.IsNullOrEmpty(value))
             {
-                return null;
+                return;
             }
 
-            if (!value.Contains("://"))
+            if (value.IndexOfAny(s_proxyDelimiters) == -1)
             {
-                value = "http://" + value;
+                ParseProxyConfigPart(value, ref insecureProxy, ref secureProxy);
+                return;
             }
 
-            if (Uri.TryCreate(value, UriKind.Absolute, out Uri uri) &&
-                    (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps))
+            string[] parts = value.Split(s_proxyDelimiters, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Could you add some comments?  It looks like the IndexOfAny block above could be removed functionality but you're doing it to avoid a string[1] allocation in the case where there aren't any separators?

Reply:
@stephentoub I imagine the most common case here is a user only has a single proxy, so I think it's an okay optimization to make.

Reply:
Agreed; I didn't realize until I read further on that this is executed for every request if there's a PAC file :(